packages/cache/src/internal/requestUtils.ts

```diff
@@ -0,0 +1,101 @@
+import * as core from '@actions/core'
+import {HttpCodes} from '@actions/http-client'
+import {
+  IHttpClientResponse,
+  ITypedResponse
+} from '@actions/http-client/interfaces'
+
+export function isSuccessStatusCode(statusCode?: number): boolean {
+  if (!statusCode) {
+    return false
+  }
+  return statusCode >= 200 && statusCode < 300
+}
+
+export function isServerErrorStatusCode(statusCode?: number): boolean {
+  if (!statusCode) {
+    return true
+  }
+  return statusCode >= 500
+}
+
+export function isRetryableStatusCode(statusCode?: number): boolean {
+  if (!statusCode) {
+    return false
+  }
+  const retryableStatusCodes = [
+    HttpCodes.BadGateway,
```

Comment:
Do we get throttled here? Should we retry on `429`?

Reply:
I don't think we should retry on throttling - it probably will make it worse, no?  :D

Reply:
(Note that I didn't add this code, just moved it to a different file for testing purposes)

I believe it's this way because a 429 would generally mean we need to wait for some period of time. The question then becomes whether it's better for us to wait and retry, or just "cache miss" and let the user rebuild the cache content.  These retries are only used when accessing the ArtifactCache API, so this would mean RU is throttling the host and that host would likely remain throttled for some time.